src/components/templates/PartyMap/components/PortalModal/PortalModal.tsx

```diff
@@ -0,0 +1,217 @@
+
+  const { enterRoom, portalVenueId } = useRoom({
+    room: portal,
+  });
+
+  const analytics = useAnalytics({ venue });
+
+  const portalVenue = findVenueInRelatedVenues(portalVenueId);
+
+  const portalVenueSubtitle = portalVenue?.config?.landingPageConfig?.subtitle;
+  const portalVenueDescription =
+    portalVenue?.config?.landingPageConfig?.description;
+
+  const [enterWithSound] = useCustomSound(portal.enterSound, {
+    interrupt: true,
+    onend: enterRoom,
+  });
+
+  // note: this is here just to change the type on it in an easy way
+  const enter: () => void = useCallback(() => {
+    analytics.trackEnterRoomEvent(portal.title, portal.template);
+    void (isExternalPortal(portal) ? openUrl(portal.url) : enterWithSound());
+  }, [analytics, enterWithSound, portal]);
+
+  const showPortalEvents = world?.showSchedule && venueEvents.length > 0;
+
+  const iconStyles = {
+    backgroundImage: portal.image_url ? `url(${portal.image_url})` : undefined,
+  };
+
+  const portalTitle = portal.title || portalVenue?.name;
+  const portalSubtitle = portal.subtitle || portalVenueSubtitle;
+  const portalDescription = portal.about || portalVenueDescription;
+
+  useEffect(() => {
+    analytics.trackOpenPortalModalEvent(portalTitle);
+  }, [analytics, portalTitle]);
+
+  // @debt maybe refactor this, but autoFocus property working very bad.
+  const enterButtonref = useRef<HTMLButtonElement>(null);
+  useEffect(() => enterButtonref.current?.focus());
+
+  return (
+    <>
+      <div className="PortalModal__main">
+        <div className="PortalModal__icon" style={iconStyles} />
+
+        <div className="PortalModal__content">
+          <div className="PortalModal__title">{portalTitle}</div>
+
+          {portalSubtitle && (
+            <ShowMoreText
+              lines={2}
+              more="Show more"
+              less="Show less"
+              className="PortalModal__subtitle"
+              expanded={false}
+              truncatedEndingComponent={"... "}
+            >
+              {portalSubtitle}
+            </ShowMoreText>
+          )}
+
+          <UserList
+            containerClassName="PortalModal__userlist"
+            usersSample={portalVenue?.recentUsersSample ?? ALWAYS_EMPTY_ARRAY}
+            userCount={portalVenue?.recentUserCount ?? 0}
+            activity={`in this ${ROOM_TAXON.lower}`}
+            attendeesTitle={world?.attendeesTitle}
+          />
+        </div>
+      </div>
+
+      {portalDescription && (
+        <ShowMoreText
+          lines={3}
+          more="Show more"
+          less="Show less"
+          className="PortalModal__description"
+          expanded={false}
+          truncatedEndingComponent={"... "}
+        >
+          <RenderMarkdown text={portalDescription} />
+        </ShowMoreText>
+      )}
+
+      <div className="PortalModal__btn-wrapper">
+        {/* @debt extract this 'enter room' button/link concept into a reusable component */}
```

Comment:
'enter room' -> 'enter portal'
Let's try and remove the word room if we can

Reply:
I may suggest `ROOM_TAXON.lower` it's clunky, but at least we have a reference to all user facing concepts that we can use to quickly find them or modify it.

Reply:
done here: https://github.com/sparkletown/sparkle/pull/2588/commits/ebbc2894098218867a7a3a8ebe54bae6b0732396